core/src/main/java/com/linecorp/armeria/internal/common/DefaultCancellationScheduler.java

```diff
@@ -292,6 +294,18 @@ public long timeoutNanos() {
         return timeoutNanos == Long.MAX_VALUE ? 0 : timeoutNanos;
     }
 
+    @Override
+    public long remainingTimeoutNanos() {
+        if (timeoutNanos == Long.MAX_VALUE) {
+            return 0;
+        }
+        if (!isStarted()) {
```

Comment:
Question) `lock` is used for thread-safety in `DefaultCancellationScheduler`, but is `remainingTimeoutNanos` unnecessary to use `lock`?

Reply:
For our purpose, I didn't think a lock is necessary since the scheduler is started either 1) before the decorator chain is triggered 2) or after the decorator chain is fully run. Meanwhile, `remainingTimeoutNanos` method is not a public method and our usage guarantees it is run during the decorator chain.

Having said this, I think in the context of `DefaultCancellationScheduler` itself, this logic may cause confusion to maintainers and locking is a small price to pay. Updated.

Reply:
I don't think there was a problem with the logic; rather, it was confusion with consistency in implementation. The new change looks good. 👍